src/lib.rs

```diff
@@ -1385,6 +1391,121 @@ where
     Some(Ratio::new(n1, d1))
 }
 
+#[cfg(all(feature = "bigint", feature = "std"))]
+impl<T: Clone + Integer + Signed + ToPrimitive + ToBigInt> ToPrimitive for Ratio<T> {
```

Comment:
It would be nice if we didn't require `Signed`, which I think is doable.

Even better if we could do it without requiring "bigint" -- do you think that's possible? What are the limiting factors? Sorry if we discussed this before, but this PR has been flushed from my brain cache... I guess the challenge with bounded types is something like `(MAX - 1) / MAX` -- you really need to promote to a larger type to be able to shift to get enough quotient bits.

Maybe we can split this up with careful cfg to avoid overlapping impls. When we don't have "bigint", we can still use the fast track for primitive integers up to 32-bit, which infallibly convert to `f64`. With `has_i128`, 64-bit integers can be implemented by going up to 128-bit to shift those extra quotient bits. Finally, when we do have "bigint", `cfg`-out the other impls and just use your generic `ToBigInt` impl.

How does that sound?

Reply:
Not requiring `Signed` seems doable, because we can always convert `T` to a `BigInt` and then ask questions about signs. I'm not sure it's doable while keeping the current optimizations on lines 1429-1440, because they rely on knowing the sign of `T` if it's signed.

For bigint, that seems feasible, although I think it would be a bit surprising to users that the performance on `Ratio<i32>` depends on whether the seemingly unrelated bigint feature is built in (presumably using `BigInt`s will be much slower than using `i64`s).